src/fparser/two/tests/fortran2003/test_defined_binary_op_r723.py

```diff
@@ -0,0 +1,79 @@
+# under the following license:
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Test Fortran 2003 rule R723 : This file tests the support for a
+Defined Binary Operator.
+
+'''
+
+import pytest
+from fparser.two.Fortran2003 import Defined_Binary_Op
+from fparser.two.utils import NoMatchError
+
+
+def test_defined_binary_operator(f2003_create):
+    '''Check that a basic binary operator is parsed correctly.
+
+    '''
+    for line in [".myoperator.",
+                 "  .myoperator.  ",
+                 ".aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+                 "aaaaaa."]:
+        ast = Defined_Binary_Op(line)
+        target = line.strip().upper()
+        assert target in str(ast)
+        assert repr(ast) == "Defined_Op('{0}')".format(target)
+
+
+def test_syntax_error(f2003_create):
+    ''' Test that NoMatchError is raised for various syntax errors. '''
+    for line in ["", "  ", "x", ".x", "x.", "..", ".,.", ". x.", ".x .",
+                 ".x x."]:
+        with pytest.raises(NoMatchError) as excinfo:
+            _ = Defined_Binary_Op(line)
+        assert "Defined_Binary_Op: '{0}'".format(line) in str(excinfo)
+
+
+def test_c703(f2003_create):
+    '''Test that we do not match if the name is more than 63 characters
+    long, the name matches an existing intrinsic operator (e.g. .gt.),
+    or the name matches an existing logical literal constant
+    (e.g. .false.).
+
+    '''
+    for line in [".aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```

Comment:
Again, can you simplify the construction of this long string?

Reply:
Done.